test/ipns-pubsub.js

```diff
@@ -0,0 +1,172 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+
+const { fromB58String } = require('multihashes')
+const base64url = require('base64url')
+const ipns = require('ipns')
+
+const parallel = require('async/parallel')
+const retry = require('async/retry')
+const series = require('async/series')
+
+const DaemonFactory = require('ipfsd-ctl')
+
+const waitFor = require('./utils/wait-for')
+
+const config = {
+  Addresses: {
+    API: '/ip4/0.0.0.0/tcp/0',
+    Gateway: '/ip4/0.0.0.0/tcp/0',
+    Swarm: []
+  }
+}
+
+const namespace = '/record/'
+
+const spawnJsDaemon = (callback) => {
+  DaemonFactory.create({ type: 'js' })
+    .spawn({
+      disposable: true,
+      initOptions: { bits: 512 },
+      args: ['--enable-namesys-pubsub'], // enable ipns over pubsub
+      config
+    }, callback)
+}
+
+const spawnGoDaemon = (callback) => {
+  DaemonFactory.create()
+    .spawn({
+      disposable: true,
+      initOptions: { bits: 1024 },
+      args: ['--enable-namesys-pubsub'], // enable ipns over pubsub
+      config
+    }, callback)
+}
+
+const ipfsRef = '/ipfs/QmPFVLPmp9zv5Z5KUqLhe2EivAGccQW2r7M7jhVJGLZoZU'
+
+describe('ipns-pubsub', () => {
+  let nodeAId
+  let nodeBId
+  let nodes = []
+
+  // Spawn daemons
+  before(function (done) {
+    // CI takes longer to instantiate the daemon, so we need to increase the timeout
+    this.timeout(80 * 1000)
+
+    series([
+      (cb) => spawnGoDaemon(cb),
+      (cb) => spawnJsDaemon(cb)
+    ], (err, daemons) => {
+      expect(err).to.not.exist()
+      nodes = daemons
+
+      done()
+    })
+  })
+
+  // Get node ids
+  before(function (done) {
+    this.timeout(100 * 1000)
+
+    parallel([
```

Comment:
Suggestion: `async/each`

Reply:
I think it is better to use `async/parallel` here, since using `async/each` here would require an extra array to keep the data with all the ids, which I feel unnecessary just for 2 elements.